controllers/metrics.go

```diff
@@ -62,9 +68,8 @@ func (s *RollingUpgradeContext) ToStepDuration(groupName, nodeName string, stepN
 	}
 }
 
-// Node turns onto step
-func (s *RollingUpgradeContext) NodeStep(InProcessingNodes map[string]*v1alpha1.NodeInProcessing,
-	nodeSteps map[string][]v1alpha1.NodeStepDuration, groupName, nodeName string, stepName v1alpha1.RollingUpgradeStep) {
+func (s *RollingUpgradeContext) DoNodeStep(InProcessingNodes map[string]*v1alpha1.NodeInProcessing,
```

Comment:
Why not merge both NodeStep() and DoNodeStep()?  I mean, any specific reasons for why you aren't merging the DoNodeStep() into NodeStep()?

Reply:
If anywhere needs to call DoNodeStep. They have to pass the "time.Now()"

Reply:
DoNodeStep needs to parameterize the endTime